src/vma/dev/net_device_table_mgr.cpp

```diff
@@ -243,16 +243,38 @@ int net_device_table_mgr::map_net_devices()
 			continue;
 		}
 
-		int iftype = get_iftype_from_ifname(ifa->ifa_name);
 		bool valid = false;
-		if (iftype == ARPHRD_INFINIBAND) {
-			if (verify_enable_ipoib(ifa->ifa_name) && verify_mlx4_ib_device(ifa->ifa_name) && verify_ipoib_mode(ifa)) {
-				valid = true;
+		if (ifa->ifa_flags & IFF_MASTER) {
+			// this is a bond interface, find the slave
+			char active_slave_name[IFNAMSIZ * MAX_SLAVES] = {0};
+			if (!get_bond_active_slave_name(ifa->ifa_name, active_slave_name, sizeof active_slave_name)) {
+				// Close the cma_id which will not be offload
+				IF_RDMACM_FAILURE(rdma_destroy_id(cma_id)) {
+					ndtm_logerr("Failed in rdma_destroy_id (errno=%d %m)", errno);
+				} ENDIF_RDMACM_FAILURE;
+				continue;
 			}
-		} else {
-			if (verify_eth_qp_creation(ifa->ifa_name)) {
-				valid = true;
+			valid = true;
+			//go over all slaves and check preconditions
+			char * slave_name;
+			slave_name = strtok (active_slave_name," ");
+			while (slave_name != NULL)
+			{
+				char* p = strchr(slave_name, '\n');
+				if (p) *p = '\0'; // Remove the tailing 'new line" char
+				if (!check_preconditions(slave_name, ifa)) {
+					valid = false;
+					vlog_printf(VLOG_WARNING,"******************************************************\n");
+					vlog_printf(VLOG_WARNING,"* Bond %s will not be offloaded.\n", ifa->ifa_name);
+					vlog_printf(VLOG_WARNING,"* Slave %s did not meet requirements.\n", slave_name);
+					vlog_printf(VLOG_WARNING,"* Check warning messages for more information.\n");
+					vlog_printf(VLOG_WARNING,"******************************************************\n");
+					break;
```

Comment:
This function map_net_devices() calls other functions which print WARNINGs. Please keep this hierarchy by calling a wrapper function for the printing